fontforge/python.c

```diff
@@ -9462,12 +9466,96 @@ static PyObject *PyFFGlyph_preserveLayer(PyFF_Glyph *self, PyObject *args) {
 Py_RETURN( self );
 }
 
-static PyObject *PyFFGlyph_BoundingBox(PyFF_Glyph *self, PyObject *UNUSED(args)) {
+static int *LayerArgToLayer(SplineFont *sf, PyObject* layerp) {
+    int *layeri = calloc(1, sizeof(int));
+
+    if ( STRING_CHECK(layerp)) {
+        char *name;
+        PYGETSTR(layerp, name, NULL);
+        *layeri = SFFindLayerIndexByName(sf, name);
+        if ( *layeri<0 ) {
+            PyErr_Format(PyExc_ValueError, "Requested layer '%s' not found", name);
+            ENDPYGETSTR();
+            return NULL;
+        }
+        ENDPYGETSTR();
+    } else if (!PyInt_Check(layerp)) {
+        PyErr_Format(PyExc_ValueError, "First argument must be string or layer index");
+        return NULL;
+    } else {
+        *layeri = PyInt_AsLong(layerp);
+    }
+    return layeri;
+}
+
+static PyObject *PyFFGlyph_boundingBox(PyFF_Glyph *self, PyObject *UNUSED(args), PyObject *keywds) {
     DBounds bb;
+    int layeri;
 
-    SplineCharFindBounds(self->sc,&bb);
+    PyObject* layerp = NULL;
+    if (keywds != NULL) layerp = PyDict_GetItemString(keywds, "layer");
+    int* layerp2;
 
-return( Py_BuildValue("(dddd)", bb.minx,bb.miny, bb.maxx,bb.maxy ));
+    if (layerp != NULL) {
+        layerp2 = LayerArgToLayer(self->sc->parent, layerp);
+        if (layerp2 == NULL) return NULL;
+        layeri = *layerp2;
+        SplineCharLayerFindBounds(self->sc, layeri, &bb);
+    } else {
+        SplineCharFindBounds(self->sc, &bb);
+    }
+
+    return( Py_BuildValue("(dddd)", bb.minx,bb.miny, bb.maxx,bb.maxy ));
+}
+
+static PyObject* PyFF_Glyph_BoundsAt(PyCFunction bounds_func, PyFF_Glyph *self, PyObject *args, PyObject *keywds) {
+    PyObject *temp = NULL;
+    double nmin = 0, nmax = 0;
+    double tnmin = 0, tnmax = 0;
+    bool set = false;
+    int layeri;
+
+    PyFF_Contour* tempc = NULL;
+
+    PyObject* layerp = NULL;
+    if (keywds != NULL) layerp = PyDict_GetItemString(keywds, "layer");
+    int* layerp2;
+
+    if (layerp != NULL) {
+        layerp2 = LayerArgToLayer(self->sc->parent, layerp);
+        if (layerp2 == NULL) return NULL;
+        layeri = *layerp2;
+    }
+
+    int arglen = PySequence_Size(args);
+    int layer_cnt = self->sc->layer_cnt;
+    for (int i = 0; i < layer_cnt; i++) {
+        if (layerp2 != NULL) {
+            if (layeri != i) continue;
+        }
+        SplineSet* ss = LayerAllSplines(&self->sc->layers[i]);
```

Comment:
So if I'm understanding right `RefChar.layers[layer].splines` is a copy of the contours from the referenced object but already transformed? That does make things easier.

Reply:
You are indeed understanding right. :-)

Reply:
In case you want some proof of this, and for posterity, here is a test script:

```python3
import fontforge

f = fontforge.fonts()[0]
print( f["A"].xBoundsAtY(600, layer=1) )
print( f["B"].xBoundsAtY(600, layer=1) )
```

Returns:

```python3
(400.6141304347826, 463.64921465968587)
(567.7201033882473, 630.7551691344895)
```

For file:

```
tri.zip: Zip archive data, at least v2.0 to extract
```

```
UEsDBBQAAgAIABqJTFC2XJKQdwIAAMMEAAAHAAAAdHJpLnNmZG1T227iMBB991dYvLSVFmo7OIS8
cd0i0VI1pay0Wq2sxAlWjY0StxX79TtOuLYVATJnZo6Px8fJVisjp9a48TDGQYch//4gNjLGS+OU
0zKjaPqm9RdMbJTefUZXUhVrF+MnWbxpUaKR3e7KBjq+4uv0BjPCyA9ofDX2w6DlyG420rgqxi2f
abM2ZdBSSuFkhj+UW2MvbGrLQuLrtXPb+PY2ByT3SAd+blroRZaVsibGhNAOIQTNnNAqHZhCg0iC
liaTpd/vo62UqyvblJzhK5W5dYw5QYMqBTkxjiA9lvuAeUrzDpzZZOMJ52Iny5F9q5NNBDB8WkOR
vrYwPWDUYyBetqDr13I2jvFvShjFIeeYh6CX9HqYsijkYRT8QYuEnTbjo7/NYGuBibZbuTB6d8gt
K/m829p76UqVwggpqgcH3c/KHw/lEe324UvQvc1UrtKvyYBwz+V5DnsnR2CR55V0nniPHEdyrBl/
qZnDQAux3deslLng9fFFCwCXrB44VdytpchODE34KX3W38QXC7xIk1k4i6vHfJJdofFkFuM+pYyj
iUlhLKaI8SxZRBHvtyl4QgEoZ8bJEvZgrJHIm32uKiAc/JyDH0v8ID9qX1ZorKqtFrtE/YOZtnsM
DeBODLQS9XlMlXu2jWdgWzOT2xiHXRyBa4ayUGa0FiUUMh4CghInSuchWOhMXMj94xlql1Lv8KkW
BTTefbKitxpK6rudSIfAalDO8QazPsLdwBP1sIb1EO6xAEfdfXAo1L5wYrITBQRe0YW44bm40D/0
W3Gr79Q9yby5LbCnh/qCkA4P+oR3IWhTHnYCFgLKjkvv/6uTMD969B9QSwECPwMUAAIACAAaiUxQ
tlySkHcCAADDBAAABwAAAAAAAAAAAAAApIEAAAAAdHJpLnNmZFBLBQYAAAAAAQABADUAAACcAgAA
AAA=
```

Sorry about the base64, GitHub is refusing to upload it. :roll_eyes: